lib/PlatoonService/src/V2VClient.cpp

```diff
@@ -0,0 +1,267 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  Vehicle to Vehicle (V2V) communication client.
+ * @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "V2VClient.h"
+#include <SettingsHandler.h>
+#include <Logging.h>
+#include <ArduinoJson.h>
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/* MQTT topic name for birth messages. */
+const char* V2VClient::TOPIC_NAME_BIRTH = "birth";
+
+/* MQTT topic name for will messages. */
+const char* V2VClient::TOPIC_NAME_WILL = "will";
+
+/** Default size of the JSON Document for parsing. */
+static const uint32_t JSON_DOC_DEFAULT_SIZE = 1024U;
+
+/** Buffer size for JSON serialization of birth / will message */
+static const uint32_t JSON_BIRTHMESSAGE_MAX_SIZE = 64U;
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+V2VClient::V2VClient(MqttClient& mqttClient) :
+    m_mqttClient(mqttClient),
+    m_waypointQueue(),
+    m_inputTopic(""),
+    m_outputTopic(""),
+    m_isLeader(false)
+{
+}
+
+V2VClient::~V2VClient()
+{
+}
+
+bool V2VClient::init(uint8_t platoonId, uint8_t vehicleId)
+{
+    bool    isSuccessful = false;
+    char    inputTopicBuffer[MAX_TOPIC_LENGTH];
+    char    outputTopicBuffer[MAX_TOPIC_LENGTH];
+    uint8_t outputTopicId = vehicleId + 1U;

```

Comment:
Why +1?

Reply:
Output is published to next vehicle.

Reply:
outputTopicId is the vehicleId of the follower ... could we rename it to a more appropriate name?